packages/lib-classifier/src/store/TutorialStore/TutorialStore.js

```diff
@@ -109,31 +84,33 @@ const TutorialStore = types
   }))
 
   .actions(self => {
-    function afterAttach () {
-      createWorkflowObserver()
-      createUPPObserver()
+    function _onUPPLoaded() {
+      const upp = getRoot(self).userProjectPreferences
+      if (upp.loadingState === asyncStates.success) {
+        self.showTutorialInModal()
+      }
     }
 
-    function createWorkflowObserver () {
-      const workflowDisposer = autorun(() => {
-        const workflow = tryReference(() => getRoot(self).workflows.active)
-        if (workflow) {
-          self.reset()
-          self.resetSeen()
-          self.fetchTutorials()
-        }
-      }, { name: 'Tutorial Store Workflow Observer autorun' })
-      addDisposer(self, workflowDisposer)
+    function _onTutorialChange() {
+      const workflow = tryReference(() => getRoot(self).workflows.active)
+      if (workflow) {
+        const { tutorial } = workflow
+        self.setActiveTutorial(tutorial?.id)
+      }
     }
 
-    function createUPPObserver () {
-      const uppDisposer = autorun(() => {
-        const upp = getRoot(self).userProjectPreferences
-        if (upp.loadingState === asyncStates.success) {
-          self.showTutorialInModal()
-        }
-      }, { name: 'Tutorial Store UPP Observer autorun' })
-      addDisposer(self, uppDisposer)
+    function _onWorkflowChange() {
+      const workflow = tryReference(() => getRoot(self).workflows.active)
+      if (workflow?.tutorials.length === 0) {
+        self.resetSeen()
+        self.fetchTutorials()
+      }
+    }
```

Comment:
I think this is happening when `useSWR` refreshes the workflow in the background, which would reset the `tutorials` array. I need to look into a bit more. You’re right that we shouldn’t need to fetch the tutorials if we already have them.

Reply:
Panoptes workflows have `workflows.links.tutorials`. We could use that to validate the tutorial IDs in the store before fetching them.

Reply:
`self.fetchTutorials()` is still being unnecessarily called as I described above. Does it make the most sense to address it in this PR or in further work with SWR?

Reply:
Right now, removing this would prevent you from refreshing a tutorial when you're editing it in the project builder. We do need some validation, to check whether a resource is already stored and possibly needs to be refreshed, before fetching data. I'm not sure how that would work with the current `Resource` and `ResourceStore` model.

Reply:
I've rewritten the data-fetching function, in #2755, so that it doesn't run when `workflowID` is undefined. That might fix this problem too, if it's being caused by the fetch running with an undefined workflow.